### ISSUE TITLE
--create-custom-diff seems to not work with scoped packages

### ISSUE THREAD

**Issue by rwwagner90** [ember-cli/ember-cli-update#464]
If you are submitting a bug report, please run your command with the debug flag:

* Unix (global): `DEBUG=ember-cli-update,boilerplate-update,git-diff-apply ember-cli-update ...`
* Windows (global): `set DEBUG=ember-cli-update,boilerplate-update,git-diff-apply && ember-cli-update ...`
* Unix (command): `DEBUG=ember-cli-update,boilerplate-update,git-diff-apply ember update ...`
* Windows (command): `set DEBUG=ember-cli-update,boilerplate-update,git-diff-apply && ember update ...`

and post the console output in the block below:

```
env DEBUG=ember-cli-update,boilerplate-update,git-diff-apply ember-cli-update --create-custom-diff
  ember-cli-update npm info ember-cli versions --json +0ms
  ember-cli-update [
  ember-cli-update   "0.0.0",
  ember-cli-update   "0.0.1",
  ember-cli-update   "0.0.2",
  ember-cli-update   "0.0.3",
  ember-cli-update   "0.0.4",
  ember-cli-update   "0.0.5",
  ember-cli-update   "0.0.6",
  ember-cli-update   "0.0.7",
  ember-cli-update   "0.0.8",
  ember-cli-update   "0.0.9",
  ember-cli-update   "0.0.10",
  ember-cli-update   "0.0.11",
  ember-cli-update   "0.0.12",
  ember-cli-update   "0.0.13",
  ember-cli-update   "0.0.14",
  ember-cli-update   "0.0.15",
  ember-cli-update   "0.0.16",
  ember-cli-update   "0.0.18",
  ember-cli-update   "0.0.19",
  ember-cli-update   "0.0.20",
  ember-cli-update   "0.0.21",
  ember-cli-update   "0.0.22",
  ember-cli-update   "0.0.23",
  ember-cli-update   "0.0.24",
  ember-cli-update   "0.0.25",
  ember-cli-update   "0.0.27",
  ember-cli-update   "0.0.28",
  ember-cli-update   "0.0.29",
  ember-cli-update   "0.0.32",
  ember-cli-update   "0.0.33",
  ember-cli-update   "0.0.34",
  ember-cli-update   "0.0.35",
  ember-cli-update   "0.0.36",
  ember-cli-update   "0.0.37",
  ember-cli-update   "0.0.38",
  ember-cli-update   "0.0.39",
  ember-cli-update   "0.0.40",
  ember-cli-update   "0.0.41",
  ember-cli-update   "0.0.42",
  ember-cli-update   "0.0.43",
  ember-cli-update   "0.0.44",
  ember-cli-update   "0.0.45",
  ember-cli-update   "0.0.46",
  ember-cli-update   "0.0.47",
  ember-cli-update   "0.1.0",
  ember-cli-update   "0.1.1",
  ember-cli-update   "0.1.2",
  ember-cli-update   "0.1.3",
  ember-cli-update   "0.1.4",
  ember-cli-update   "0.1.5",
  ember-cli-update   "0.1.6",
  ember-cli-update   "0.1.7",
  ember-cli-update   "0.1.8",
  ember-cli-update   "0.1.9",
  ember-cli-update   "0.1.10",
  ember-cli-update   "0.1.11",
  ember-cli-update   "0.1.12",
  ember-cli-update   "0.1.13",
  ember-cli-update   "0.1.14",
  ember-cli-update   "0.1.15",
  ember-cli-update   "0.2.0-beta.1",
  ember-cli-update   "0.2.0",
  ember-cli-update   "0.2.1",
  ember-cli-update   "0.2.2",
  ember-cli-update   "0.2.3",
  ember-cli-update   "0.2.4",
  ember-cli-update   "0.2.5",
  ember-cli-update   "0.2.6",
  ember-cli-update   "0.2.7",
  ember-cli-update   "1.13.0",
  ember-cli-update   "1.13.1",
  ember-cli-update   "1.13.5",
  ember-cli-update   "1.13.6",
  ember-cli-update   "1.13.7",
  ember-cli-update   "1.13.8",
  ember-cli-update   "1.13.9",
  ember-cli-update   "1.13.10",
  ember-cli-update   "1.13.11",
  ember-cli-update   "1.13.12",
  ember-cli-update   "1.13.13",
  ember-cli-update   "1.13.14",
  ember-cli-update   "1.13.15",
  ember-cli-update   "2.2.0-beta.1",
  ember-cli-update   "2.2.0-beta.2",
  ember-cli-update   "2.2.0-beta.3",
  ember-cli-update   "2.2.0-beta.4",
  ember-cli-update   "2.2.0-beta.5",
  ember-cli-update   "2.2.0-beta.6",
  ember-cli-update   "2.3.0-beta.1",
  ember-cli-update   "2.3.0-beta.2",
  ember-cli-update   "2.3.0",
  ember-cli-update   "2.4.0",
  ember-cli-update   "2.4.1",
  ember-cli-update   "2.4.2",
  ember-cli-update   "2.4.3",
  ember-cli-update   "2.5.0",
  ember-cli-update   "2.5.1",
  ember-cli-update   "2.6.0-beta.1",
  ember-cli-update   "2.6.0-beta.2",
  ember-cli-update   "2.6.0-beta.3",
  ember-cli-update   "2.6.0",
  ember-cli-update   "2.6.1",
  ember-cli-update   "2.6.2",
  ember-cli-update   "2.6.3",
  ember-cli-update   "2.7.0-beta.1",
  ember-cli-update   "2.7.0-beta.2",
  ember-cli-update   "2.7.0-beta.3",
  ember-cli-update   "2.7.0-beta.4",
  ember-cli-update   "2.7.0-beta.5",
  ember-cli-update   "2.7.0-beta.6",
  ember-cli-update   "2.7.0",
  ember-cli-update   "2.8.0-beta.1",
  ember-cli-update   "2.8.0-beta.2",
  ember-cli-update   "2.8.0-beta.3",
  ember-cli-update   "2.8.0",
  ember-cli-update   "2.9.0-beta.1",
  ember-cli-update   "2.9.0-beta.2",
  ember-cli-update   "2.9.0",
  ember-cli-update   "2.9.1",
  ember-cli-update   "2.10.0-beta.1",
  ember-cli-update   "2.10.0-beta.2",
  ember-cli-update   "2.10.0",
  ember-cli-update   "2.10.1",
  ember-cli-update   "2.11.0-beta.1",
  ember-cli-update   "2.11.0-beta.2",
  ember-cli-update   "2.11.0",
  ember-cli-update   "2.11.1",
  ember-cli-update   "2.12.0-beta.1",
  ember-cli-update   "2.12.0-beta.2",
  ember-cli-update   "2.12.0",
  ember-cli-update   "2.12.1",
  ember-cli-update   "2.12.2",
  ember-cli-update   "2.12.3",
  ember-cli-update   "2.13.0-beta.1",
  ember-cli-update   "2.13.0-beta.2",
  ember-cli-update   "2.13.0-beta.3",
  ember-cli-update   "2.13.0-beta.4",
  ember-cli-update   "2.13.0",
  ember-cli-update   "2.13.1",
  ember-cli-update   "2.13.2",
  ember-cli-update   "2.13.3",
  ember-cli-update   "2.14.0-beta.1",
  ember-cli-update   "2.14.0-beta.2",
  ember-cli-update   "2.14.0",
  ember-cli-update   "2.14.1",
  ember-cli-update   "2.14.2",
  ember-cli-update   "2.15.0-beta.1",
  ember-cli-update   "2.15.0-beta.2",
  ember-cli-update   "2.15.0",
  ember-cli-update   "2.15.1",
  ember-cli-update   "2.16.0-beta.1",
  ember-cli-update   "2.16.0-beta.2",
  ember-cli-update   "2.16.0",
  ember-cli-update   "2.16.1",
  ember-cli-update   "2.16.2",
  ember-cli-update   "2.17.0-beta.1",
  ember-cli-update   "2.17.0-beta.2",
  ember-cli-update   "2.17.0",
  ember-cli-update   "2.17.1",
  ember-cli-update   "2.17.2",
  ember-cli-update   "2.18.0-beta.1",
  ember-cli-update   "2.18.0-beta.2",
  ember-cli-update   "2.18.0",
  ember-cli-update   "2.18.1",
  ember-cli-update   "2.18.2",
  ember-cli-update   "3.0.0-beta.1",
  ember-cli-update   "3.0.0-beta.2",
  ember-cli-update   "3.0.0",
  ember-cli-update   "3.0.1",
  ember-cli-update   "3.0.2",
  ember-cli-update   "3.0.3",
  ember-cli-update   "3.0.4",
  ember-cli-update   "3.1.0-beta.1",
  ember-cli-update   "3.1.0",
  ember-cli-update   "3.1.1",
  ember-cli-update   "3.1.2",
  ember-cli-update   "3.1.3",
  ember-cli-update   "3.1.4",
  ember-cli-update   "3.2.0-beta.1",
  ember-cli-update   "3.2.0-beta.2",
  ember-cli-update   "3.2.0",
  ember-cli-update   "3.3.0",
  ember-cli-update   "3.4.0-beta.1",
  ember-cli-update   "3.4.0-beta.2",
  ember-cli-update   "3.4.0-beta.3",
  ember-cli-update   "3.4.1",
  ember-cli-update   "3.4.2-beta.1",
  ember-cli-update   "3.4.2",
  ember-cli-update   "3.4.3",
  ember-cli-update   "3.4.4",
  ember-cli-update   "3.5.0-beta.1",
  ember-cli-update   "3.5.0-beta.2",
  ember-cli-update   "3.5.0",
  ember-cli-update   "3.5.1",
  ember-cli-update   "3.6.0-beta.1",
  ember-cli-update   "3.6.0",
  ember-cli-update   "3.7.0-beta.1"
  ember-cli-update ]
  ember-cli-update  +662ms
  ember-cli-update npm info ember-cli@latest version --json +5ms
  ember-cli-update "3.6.0"
  ember-cli-update  +555ms
  ember-cli-update node /Users/rwwagner/html-next/flexi-config/node_modules/ember-cli/bin/ember addon @html-next/flexi-config -sn -sg +7ms
ENOENT: no such file or directory, mkdir '@html-next/flexi-config'


Stack Trace and Error Report: /var/folders/tf/h_tyhd41165d978ps4hgc5wc0000gn/T/error.dump.344717bf808e41eb4458fbaeb2a38eb7.log
{ Error: Command failed: node /Users/rwwagner/html-next/flexi-config/node_modules/ember-cli/bin/ember addon @html-next/flexi-config -sn -sg
ENOENT: no such file or directory, mkdir '@html-next/flexi-config'


Stack Trace and Error Report: /var/folders/tf/h_tyhd41165d978ps4hgc5wc0000gn/T/error.dump.344717bf808e41eb4458fbaeb2a38eb7.log

    at checkExecSyncError (child_process.js:621:11)
    at Object.execSync (child_process.js:658:13)
    at Object.run (/Users/rwwagner/.nvm/versions/node/v11.3.0/lib/node_modules/ember-cli-update/src/run.js:8:19)
    at tmpDir.then.cwd (/Users/rwwagner/.nvm/versions/node/v11.3.0/lib/node_modules/ember-cli-update/src/get-start-and-end-commands.js:63:13)
  status: 1,
  signal: null,
  output:
   [ null,
     <Buffer 57 41 52 4e 49 4e 47 3a 20 4e 6f 64 65 20 76 31 31 2e 33 2e 30 20 69 73 20 6e 6f 74 20 74 65 73 74 65 64 20 61 67 61 69 6e 73 74 20 45 6d 62 65 72 20 ... 135 more bytes>,
     <Buffer 45 4e 4f 45 4e 54 3a 20 6e 6f 20 73 75 63 68 20 66 69 6c 65 20 6f 72 20 64 69 72 65 63 74 6f 72 79 2c 20 6d 6b 64 69 72 20 27 40 68 74 6d 6c 2d 6e 65 ... 146 more bytes> ],
  pid: 12074,
  stdout:
   <Buffer 57 41 52 4e 49 4e 47 3a 20 4e 6f 64 65 20 76 31 31 2e 33 2e 30 20 69 73 20 6e 6f 74 20 74 65 73 74 65 64 20 61 67 61 69 6e 73 74 20 45 6d 62 65 72 20 ... 135 more bytes>,
  stderr:
   <Buffer 45 4e 4f 45 4e 54 3a 20 6e 6f 20 73 75 63 68 20 66 69 6c 65 20 6f 72 20 64 69 72 65 63 74 6f 72 79 2c 20 6d 6b 64 69 72 20 27 40 68 74 6d 6c 2d 6e 65 ... 146 more bytes> }
  ember-cli-update npx -p ember-cli@3.6.0 ember addon @html-next/flexi-config -sn -sg +0ms
../fsevents.cc:63:6: warning: field 'async_resource' will be initialized after field 'lockStarted' [-Wreorder]
   : async_resource("fsevents:FSEvents"), lockStarted(false) {
     ^
1 warning generated.
WARNING: Node v11.3.0 is not tested against Ember CLI on your platform. We recommend that you use the most-recent "Active LTS" version of Node.js. See https://git.io/v7S5n for details.
ENOENT: no such file or directory, mkdir '@html-next/flexi-config'


Stack Trace and Error Report: /var/folders/tf/h_tyhd41165d978ps4hgc5wc0000gn/T/error.dump.4d832f352f205afb61078d9b222a2d2c.log
```

**Comment by kellyselden** [comment]
Makes sense. Can you share with me how you created this addon? I'm not familiar. You didn't run `ember addon @html-next/flexi-config` did you? Did you run `ember addon flexi-config` and then tweak the files?

**Comment by rwwagner90** [comment]
@kellyselden yeah, these packages were not originally scoped to `@html-next`. They were just `flexi-config` etc.

**Comment by kellyselden** [comment]
This https://github.com/ember-cli/ember-cli-update/blob/d612f6ae0e4377bd9af271cd1c1c774da2d1b189/src/get-start-and-end-commands.js#L52 and https://github.com/ember-cli/ember-cli-update/blob/d612f6ae0e4377bd9af271cd1c1c774da2d1b189/src/get-start-and-end-commands.js#L67 need more code to handle scopes. (and/or ember-cli can update to create scoped addons without tweaks.)

**Comment by kellyselden** [comment]
upstream tracking here https://github.com/ember-cli/ember-cli/issues/4372

**Comment by kellyselden** [comment]
Turns out this can work on any ember-cli version with --directory.

**Comment by rwjblue** [comment]
ember-cli@3.15.0 was released yesterday (3.15.1 is out today :sob: ), and `ember addon @foo/bar` is now supported out of the box. 

I think that should resolve this issue (at least going _forward_), but might need to use `--directory` for generating the diff on older versions :thinking:
